src/microbe_stage/ProcessSystem.cs

```diff
@@ -49,14 +49,17 @@ public ProcessSystem(Node worldRoot)
     public static EnergyBalanceInfo ComputeEnergyBalance(IEnumerable<OrganelleTemplate> organelles,
         BiomeConditions biome, MembraneType membrane)
     {
-        return ComputeEnergyBalance(organelles.Select(o => o.Definition), biome, membrane);
+        organelles = organelles.ToList();
+
+        var maximumMovementDirection = MicrobeInternalCalculations.MaximumSpeedDirection(organelles);
+        return ComputeEnergyBalance(organelles, biome, membrane, maximumMovementDirection);
     }
 
     /// <summary>
     ///   Computes the energy balance for the given organelles in biome
     /// </summary>
-    public static EnergyBalanceInfo ComputeEnergyBalance(IEnumerable<OrganelleDefinition> organelles,
-        BiomeConditions biome, MembraneType membrane)
+    public static EnergyBalanceInfo ComputeEnergyBalance(IEnumerable<OrganelleTemplate> organelles,
+        BiomeConditions biome, MembraneType membrane, Vector3 maximumMovementDirection)
```

Comment:
`maximumMovementDirection` should be renamed here to something that explains it is used to filter out movementOrganelles that aren't active if they don't allow moving in that direction. Maybe something like `excludeMovementIfCannotPushInThisDirection` though that's a bit wordy

Reply:
why should i rename  a variable like that? thats not what the variable represents. Armed with some basic vector math and seeing its use in calcualtions you should know in what way and why the variable is used.

Reply:
It doesn't fit the design of ComputeEnergyBalance. It should not have to know that maximum movement direction is the only direction flagella are ever used in. This is a lower level building block that should be made with the assumption that whoever is looking at it doesn't know anything about maximum movement direction, just that if you want this lower level method can exclude any movement organelles that can't move the cell in a certain direction from the computation.

Reply:
Totally disagree, when computing the maximum energy consumption of a microbe the direction in which the speed and, thus, the amount of flagella activation is the greatest is pretty relevant since both are tied together.

Reply:
I would be more willing to accept that, if this method wasn't public. This is a public method and now it has this new confusing parameter that doesn't seem to be related to this method's purpose at all. I might accept this naming if you add xml parameter comments.

Reply:
For all the parameters?

Reply:
I'll do this myself.